mrp_bom_enforcement/views/mrp_production_views.xml

```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<odoo>
+    <record id="mrp_production_form_view_inherit" model="ir.ui.view">
+        <field name="name">mrp.production.form.inherit</field>
+        <field name="model">mrp.production</field>
+        <field name="inherit_id" ref="mrp.mrp_production_form_view" />
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='move_raw_ids']" position="attributes">
+                <attribute name="attrs">{
+                    'readonly': ['|', '|', ('bom_id', '=', False),
+                    ('state', '=', 'cancel'),
+                    '&amp;', ('state', '=', 'done'),
+                    ('is_locked', '=', True)]
+                }</attribute>
```

Comment:
Not comfortable with this domain override.  If we are to do this, we may consider using base_view_inheritance_extension, however I actually prefer not to change anything on the view.  Can we achieve the goal by adding a constraint method instead?